custom/icds/messaging/custom_content.py

```diff
@@ -263,16 +268,38 @@ def aww_1(recipient, case_schedule_instance):
 
 def aww_2(recipient, case_schedule_instance):
     indicator_class = AWWAggregatePerformanceIndicator
-    if _use_v2_indicators(case_schedule_instance.case):
+    if _use_v2_indicators(case_schedule_instance.case, AWW_APP_ID):
         indicator_class = AWWAggregatePerformanceIndicatorV2
     return run_indicator_for_usercase(case_schedule_instance.case, indicator_class)
 
 
-def _use_v2_indicators(usercase):
+def _use_v2_indicators(usercase, app_id):
+    app_version_in_use = _get_app_version_used_by_usercase(app_id, usercase)
+    return app_version_in_use and _has_functional_version_set(usercase.domain, app_id, app_version_in_use)
+
+
+def _get_app_version_used_by_usercase(app_id, usercase):
     if usercase.type != USERCASE_TYPE:
         raise ValueError("Expected '%s' case" % USERCASE_TYPE)
-    # ToDo: add condition to check for functional version of the app
-    return True
+    user = get_user_from_usercase(usercase)
+    last_build_details = _get_reported_last_build_of_app_by_user(app_id, user)
+    if last_build_details:
+        return last_build_details.build_version
```

Comment:
does that violate "more than one exit for a function" guideline?

Reply:
No. It's fine to use multiple return statements as long as program flow is not hard to follow. One example where it is a bad idea is in very long functions (but those are discouraged for other reasons anyway).

Reply:
that is good to understand. I definitely favour being explicit.
When it comes to `return None`, these are also called out to be `redundant` because that is a default return anyway but again explicit is good.

Reply:
I always `return None` (rather than using an implicit return) when a function has any other return statement that returns a value other than `None` because "explicit is better than implicit" and it makes it harder for reviewers to tell if you made a coding error (forgot to return something).